### PR TITLE
Enable Whisper English Translation

### PR DESCRIPTION
**Pull request by mliradelc**
Ready for review.

I left config files ready for testing the PR. at the time is approved I will deleted those modificactions. If you don't wan't I can delete them immediately.

Enables Whisper's translation to english.

To make it compatible, I had to add Vosk a new setting `vosk.default.language` Because needs a target language to work and Whisper needs to be empty to avoid wrong language target.

TO DO before lift draft status:

* [x] Write Documentation
* [ ] Delete transient settings (Used for speed up testing and development)

Other

* [x] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

File: etc/org.opencastproject.speechtotext.impl.SpeechToTextServiceImpl.cfg [hunk 1]
```diff
@@ -4,4 +4,4 @@
```
Comment (mliradelc) [inline etc/org.opencastproject.speechtotext.impl.SpeechToTextServiceImpl.cfg]: Transient Parameter

File: etc/org.ops4j.pax.logging.cfg [hunk 2]
```diff
@@ -24,6 +24,9 @@ log4j2.rootLogger.appenderRef.Console.filter.threshold.level = ${karaf.log.conso
 # detail level of logs, you can adjust this to: ERROR, WARN, INFO, DEBUG, TRACE.
 log4j2.logger.opencast.name = org.opencastproject
 log4j2.logger.opencast.level = INFO
+log4j2.logger.stt.name = org.opencastproject.speechtotext
+log4j2.logger.stt.level = DEBUG
+
```
Comment (mliradelc) [inline etc/org.ops4j.pax.logging.cfg]: Transient parameters

File: etc/workflows/fast.yaml [hunk 3]
```diff
@@ -34,6 +34,17 @@ operations:
...
+    fail-on-error: true
+    exception-handler-workflow: "partial-error"
+    description: "Generating subtitles with Whisper"
+    configurations:
+      - source-flavor: "presenter/source"
+      - target-flavor: "captions/vtt"
+      - target-tags: "archive,engage-download,lang:en"
+      - target-element: "track"
+      - translate: "true"
+
```
Comment (mliradelc) [inline etc/workflows/fast.yaml]: Transient Parameters

File: etc/workflows/fast.yaml [hunk 4]
```diff
@@ -34,6 +34,17 @@ operations:
       - overwrite: false
       - accept-no-media: false
 
+  - id: speechtotext
```
Comment (gregorydlogan) [inline etc/workflows/fast.yaml]: I know `fast` is supposed to be mainly for testing, but is this going to fail if Whisper/Vosk isn't installed?

File: modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/VoskEngine.java [hunk 5]
```diff
@@ -120,8 +139,11 @@ public File generateSubtitlesFile(File mediaFile, File preparedOutputFile, Strin
     } finally {
       IoSupport.closeQuietly(process);
     }
+    List<Object> returnValues = new ArrayList<>();
```
Comment (gregorydlogan) [inline modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/VoskEngine.java]: Why not a `Map<String, File>` where the key is the lang?

File: modules/speech-to-text-workflowoperation/src/main/java/org/opencastproject/workflow/handler/speechtotext/SpeechToTextWorkflowOperationHandler.java [hunk 6]
```diff
@@ -158,18 +165,19 @@ public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobConte
    * @param parentMediaPackage The media package where the track is located.
    * @param tagsAndFlavors Tags and flavors instance (to get target flavor information)
    * @param appendSubtitleAs Tells how the subtitles file has to be appended.
+   * @param translate Allows
```
Comment (gregorydlogan) [inline modules/speech-to-text-workflowoperation/src/main/java/org/opencastproject/workflow/handler/speechtotext/SpeechToTextWorkflowOperationHandler.java]: Missed the rest of the description here.

**Review by gregorydlogan — APPROVED** [review]
Looks good to me, aside from the missed description (which could be fixed later) and the one return type.  Have not had time to test, but it certainly looks like it should work.

If there's an RM looking to merge this prior to the next 13.x release I'd say I would be comfortable merging it as is and fixing the rest in #4770 or a follow on PR.

File: etc/workflows/fast.yaml (hunk 4 above)
Comment (mliradelc) [reply etc/workflows/fast.yaml]: Indeed, but as I say before, it's only for testing this PR. after approval I will rollback this file and these other files:
- [etc/org.ops4j.pax.logging.cfg](https://github.com/opencast/opencast/pull/4738/files#diff-f815dab0a94901642bd7cd180fcb31d4dcb3ea20585887c575a687fe72f99ca9)
- [etc/org.opencastproject.speechtotext.impl.SpeechToTextServiceImpl.cfg](https://github.com/opencast/opencast/pull/4738/files#diff-44744be62e3ea85de7a6567c32939ad7faefb041c45b7e76b4e0cdbc251c7850)

File: modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/VoskEngine.java (hunk 5 above)
Comment (mliradelc) [reply modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/VoskEngine.java]: That's … a good idea! Easier to read.

Comment (mliradelc) [reply modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/VoskEngine.java]: Done!

File: modules/speech-to-text-workflowoperation/src/main/java/org/opencastproject/workflow/handler/speechtotext/SpeechToTextWorkflowOperationHandler.java (hunk 6 above)
Comment (mliradelc) [reply modules/speech-to-text-workflowoperation/src/main/java/org/opencastproject/workflow/handler/speechtotext/SpeechToTextWorkflowOperationHandler.java]: Done!

**Comment by mliradelc** [comment]
I had to make a new commit, that is a Hotfix against the new Whisper version. Whisper Changed the format filename output from `somefile.mp4.vtt` to `somefile.vtt`

I you are using whisper from some time, please update it.

File: etc/org.opencastproject.speechtotext.impl.SpeechToTextServiceImpl.cfg
Comment (mliradelc) [inline etc/org.opencastproject.speechtotext.impl.SpeechToTextServiceImpl.cfg:1]: Transient change for testing purposes only, will be deleted when approved

**Comment by mliradelc** [comment]
@gregorydlogan All set, you can merge it, the transient files have been deleted.

**Pull request merged**